ui-components/src/page_components/listing/ListingCard.tsx

```diff
@@ -22,10 +25,75 @@ export interface ListingCardProps {
   tableHeaderProps?: ListingCardHeaderProps
   tableProps: ListingCardTableProps
   detailsLinkClass?: string
+  listing: Listing
 }
 
 const ListingCard = (props: ListingCardProps) => {
   const { imageCardProps, tableProps, detailsLinkClass, tableHeaderProps } = props
+  const { profile, userProfileService } = useContext(AuthContext)
+  const [updatedFavorites, setUpdatedFavorites] = useState(profile?.preferences?.favoriteIDs)
+
+  useEffect(() => {
+    setFavoriteButtonState()
+  })
+
+  const setFavoriteButtonState = () => {
+    const preferences: UserPreferences = profile?.preferences || { favoriteIDs: [] }
+    setUpdatedFavorites(preferences?.favoriteIDs)
+  }
+
+  const addToFavorite = async () => {
+    if (!profile) {
+      return
+    }
+    const preferences: UserPreferences = profile?.preferences || { favoriteIDs: [] }
```

Comment:
You can remove the `?` here, since you're early returning if `profile` is null above.

Reply:
updated